asr1k_neutron_l3/plugins/l3/agents/asr1k_l3_agent.py

```diff
@@ -338,63 +327,70 @@ def __init__(self, host, conf=None):
         #             continue
         #     break
         self.monitor = self._initialize_monitor()
+        self.worker = periodics.PeriodicWorker([
```

Comment:
Why don't you use [`PeriodicWorker.create()`](https://github.com/openstack/futurist/blob/d17f58d1bf26fde903578260c87da44c191258ae/futurist/periodics.py#L413) here, but instantiate the worker directly?

Wouldn't it make sense to create a worker with the `ThreadPoolExecutor` instead of the default `SynchronousExecutor`? As I understand it, the previous periodic-tasks would be able to run in parallel so this would better mimic previous behavior.

Reply:
Looking further into it, `neutron.service.Service` starts 2 loopingcalls in `start()`: one for reporting state and one for running the periodic tasks synchronous. This makes sure that we always report state, even if some periodic tasks take some time. Therefore, the default `SynchronousExecutor` might be fine here, but we cannot have the `self._report_state()` handled by `self.worker`, too.